src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,527 @@
+    ) external initializer {
+        _initializePauser(_pauserRegistry, initialPausedStatus);
+        _transferOwnership(initialOwner);
+    }
+
+    /// @inheritdoc IAllocationManager
+    function slashOperator(
+        SlashingParams calldata params
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
+        require(0 < params.wadToSlash && params.wadToSlash <= WAD, InvalidWadToSlash());
+
+        // Check that the operator is registered and slashable
+        OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: params.operatorSetId});
+        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
+        require(avsDirectory.isOperatorSlashable(params.operator, operatorSet), InvalidOperator());
+
+        // Record the proportion of 1e18 that the operator's total shares that are being slashed
+        uint256[] memory wadSlashed = new uint256[](params.strategies.length);
+
+        for (uint256 i = 0; i < params.strategies.length; ++i) {
+            PendingMagnitudeInfo memory info =
+                _getPendingMagnitudeInfo(params.operator, params.strategies[i], operatorSetKey);
+
+            require(info.currentMagnitude > 0, OperatorNotAllocated());
+
+            // 1. Calculate slashing amount and update current/encumbered magnitude
+            uint64 slashedMagnitude = uint64(uint256(info.currentMagnitude).mulWad(params.wadToSlash));
+            info.currentMagnitude -= slashedMagnitude;
+            info.encumberedMagnitude -= slashedMagnitude;
+
+            // 2. If there is a pending deallocation, reduce pending deallocation proportionally.
+            // This ensures that when the deallocation is completed, less magnitude is freed.
+            if (info.pendingDiff < 0) {
+                uint64 slashedPending = uint64(uint256(uint128(-info.pendingDiff)).mulWad(params.wadToSlash));
+                info.pendingDiff += int128(uint128(slashedPending));
+
+                emit OperatorSetMagnitudeUpdated(
+                    params.operator,
+                    operatorSet,
+                    params.strategies[i],
+                    _addInt128(info.currentMagnitude, info.pendingDiff),
+                    info.effectTimestamp
+                );
+            }
+
+            // 3. Update the operator's allocation in storage
+            _updateMagnitudeInfo({
+                operator: params.operator,
+                strategy: params.strategies[i],
+                operatorSetKey: operatorSetKey,
+                info: info
+            });
+
+            emit OperatorSetMagnitudeUpdated(
+                params.operator, operatorSet, params.strategies[i], info.currentMagnitude, uint32(block.timestamp)
+            );
+
+            // 4. Reduce the operator's max magnitude
+            uint64 maxMagnitudeBeforeSlash = _maxMagnitudeHistory[params.operator][params.strategies[i]].latest();
+            uint64 maxMagnitudeAfterSlash = maxMagnitudeBeforeSlash - slashedMagnitude;
```

Comment:
Looks like slashingMagnitude is not also adding slashed amount from deallocations in step 2!
`slashedPending` amount needs to be added to `slashedMagnitude`